### ISSUE TITLE
Inconsistent treeline color when VCL Styles enabled in Delphi XE4

### ISSUE THREAD

**Issue by raelb** [JAM-Software/Virtual-TreeView#733]
I create a simple project in Delphi XE4 with `Amethyst Kamri` VCL Style enabled. `RootNodeCount = 30`.

Appearance when run the project:

![image](https://user-images.githubusercontent.com/13108198/31781952-ec315216-b4f9-11e7-8f33-fb4457e9ac52.png)
(Bright Green treeline...)

After I move the mouse over the VST, from top to bottom (no clicking...)
![image](https://user-images.githubusercontent.com/13108198/31782029-1d20da86-b4fa-11e7-8837-fbfb6acabc62.png)

Using 6.7.0

**Comment by joachimmarder** [comment]
I can't replicate this with 10.2 Tokyo. And I have XE4 no longer installed. Any chance that you try you sample project with 10.2 Tokyo?

**Comment by raelb** [comment]
Yes, this problem does not occur with Tokyo.

**Comment by joachimmarder** [comment]
Well i guess then it is a problem with XE4 or the theme. Do other themes under XE4 show the same problems, or is only `Amethyst Kamri` affected?

**Comment by raelb** [comment]
Same problem with other themes. Thankfully in my main app this problem does not occur. Looks like some TreeOption bypasses the problem.

**Comment by joachimmarder** [comment]
> Thankfully in my main app this problem does not occur. 

In this case: Can you please attach your sample project to this issue?

VCL themes were a bit buggy in the first releases, it became better with XE5 and XE6. Maybe someone still using XE4 can look into this issue.

**Comment by raelb** [comment]
I have attached a sample project. However nothing special to it. 

1. Set RootNodeCount = 30
2. Set Text in GetText
3. Set VCL Style in Project options.

Move mouse over nodes, and select different nodes.

[VTV VCL Styles Bug.zip](https://github.com/Virtual-TreeView/Virtual-TreeView/files/1404520/VTV.VCL.Styles.Bug.zip)

**Comment by joachimmarder** [comment]
> I have attached a sample project.

I don't see any problem here under 10.2 Tokyo, so most likely this issue is related to XE4 and maybe XE3.  Maybe someone still using XE4 can look into this issue.

**Comment by sanjayssk** [comment]
Well, I do have Delphi XE4 and I can't reproduce this problem. It draws lines in correct color straight out of the box. Moving mouse has no effect.

**Comment by joachimmarder** [comment]
Interesting. This bug surprised me a little, because it is so obvious that I expected it to be reported a long time ago.  @raelb : Maybe a problem with some Delphi AddOn or your machine?

**Comment by joachimmarder** [comment]
@raelb: Any news on this issue? Do you have the chance to test on another machine?

**Comment by raelb** [comment]
Yes, you are correct. It does seem to be a problem on my machine. If I run the exe (created on my machine with XE4) on another machine, there is no problem at all. 

Conversely, on the other machine I created an exe with Tokyo. It runs with no problems. If I run this exe on my machine, then there is still some visual defect, although much less than before.

![image](https://user-images.githubusercontent.com/13108198/32150410-09df97ea-bd1b-11e7-9c92-85f1950e16d1.png)

In this case there is a slight brown color that appears over the tree lines after the mouse moves over them.

I was actually experiencing a similar problem with a TRichview related component: TRVOfficeRadioButton. With styles enabled, on my machine there was a huge black area:

![image](https://user-images.githubusercontent.com/13108198/32150441-941ab4a8-bd1b-11e7-87be-104cef758f9c.png)

Although, on the other computer there was no problem. However, this could be fixed with the following change in code, suggested by the author:

> 
>In RVOfficeRadioBtn.pas, find the line
>    StyleServices.DrawParentBackground(0, Canvas.Handle, nil, False, R);
>Change it to
>    StyleServices.DrawParentBackground(Handle, Canvas.Handle, nil, False, R);
>

I did not notice VirtualTrees using 0 instead of Handle. It is strange, as I mentioned above, that enabling certain flags (e.g. hottrack) makes the problem go away.
